src/EndpointManager.sol

```diff
@@ -64,6 +78,76 @@ abstract contract EndpointManager is IEndpointManager, OwnableUpgradeable, Reent
     /// @dev This will either cross-call or internal call, depending on whether the contract is standalone or not.
     function setSibling(uint16 siblingChainId, bytes32 siblingContract) external virtual;
 
+    function setOutboundLimit(uint256 limit) external onlyOwner {
+        uint256 oldLimit = outboundLimitParams.limit;
+        uint256 currentCapacity = getCurrentOutboundCapacity();
+        outboundLimitParams.limit = limit;
+
+        outboundLimitParams.currentCapacity =
+            _calculateNewCurrentCapacity(limit, oldLimit, currentCapacity);
+
+        outboundLimitParams.ratePerSecond = limit / _RATE_LIMIT_DURATION;
+        outboundLimitParams.lastTxTimestamp = block.timestamp;
+    }
+
+    function getCurrentOutboundCapacity() public view returns (uint256) {
+        return _getCurrentCapacity(outboundLimitParams);
+    }
+
+    /**
+     * @dev Gets the current capacity for a parameterized rate limits struct
+     */
+    function _getCurrentCapacity(RateLimitParams memory rateLimitParams)
+        internal
+        view
+        returns (uint256 capacity)
+    {
+        capacity = rateLimitParams.currentCapacity;
+        if (capacity == rateLimitParams.limit) {
+            return capacity;
+        } else if (rateLimitParams.lastTxTimestamp + _RATE_LIMIT_DURATION <= block.timestamp) {
+            capacity = rateLimitParams.limit;
+        } else if (rateLimitParams.lastTxTimestamp + _RATE_LIMIT_DURATION > block.timestamp) {
+            uint256 timePassed = block.timestamp - rateLimitParams.lastTxTimestamp;
+            uint256 calculatedCapacity = capacity + (timePassed * rateLimitParams.ratePerSecond);
+            capacity = calculatedCapacity > rateLimitParams.limit
+                ? rateLimitParams.limit
+                : calculatedCapacity;
+        }
+    }
+
+    /**
+     * @dev Updates the current capacity
+     *
+     * @param newLimit The new limit
+     * @param oldLimit The old limit
+     * @param currentLimit The current limit
+     */
+    function _calculateNewCurrentCapacity(
```

Comment:
Is this method copy pasted from xERC20 or new logic? I'd like to see tests for updating the limits in both directions when the current limit isn't 100% of the oldLimit (i.e. a transfer has consumed some of the limit)

Reply:
Pretty much copy-pasted, just changed some variable names (imo using `currentCapacity` is a bit more descriptive than `currentLimit`).

I'll add tests for updating limits.

Reply:
Cool thanks. I just want to be sure that at the new rate the limits are filling to the correct max. Since it's mainly copy pasted there should hopefully be some tests from them you can repurpose as well